core/arch/arm/plat-rockchip/common.h

```diff
@@ -33,4 +33,18 @@
 #define LOCK_ADDR_OFFSET	4
 #define BOOT_ADDR_OFFSET	8
 
+/******************************************************************************
+ * The register have write-mask bits, it is mean, if you want to set the bits,
+ * you needs set the write-mask bits at the same time, the write-mask bits is
+ * in high 16-bits.
+ * The fllowing macro definition helps access write-mask bits reg efficient.
+ ******************************************************************************/
+#define REG_MSK_SHIFT		16
+#define WMSK_BIT(nr)		BIT((nr) + REG_MSK_SHIFT)
+#define BIT_WITH_WMSK(nr)	(BIT(nr) | WMSK_BIT(nr))
+#define BITS_SHIFT(bits, shift)	SHIFT_U32(bits, shift)
+#define BITS_WMSK(msk, shift)	SHIFT_U32(msk, (shift) + REG_MSK_SHIFT)
+#define BITS_WITH_WMASK(bits, msk, shift)\
+	(BITS_SHIFT(bits, shift) | BITS_SHIFT(msk, ((shift) + REG_MSK_SHIFT)))
```

Comment:
This can be rewritten in a simpler way:
```C
#define BITS_WMSK(msk, shift) SHIFT_U32(msk, (shift) + 16)
#define BITS_WITH_WMASK(bits, msk, shift) \
	(SHIFT_U32(bits, shift) | BITS_WMSK(msk, shift))
```